molecules/pubchem/pipeline.py

```diff
@@ -0,0 +1,193 @@
+# Copyright 2018 Google Inc. All Rights Reserved. Licensed under the Apache
+# License, Version 2.0 (the "License"); you may not use this file except in
+# compliance with the License. You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import sdf
+
+import json
+import logging
+import pprint
+
+import apache_beam as beam
+import tensorflow as tf
+import tensorflow_transform as tft
+
+from apache_beam.io import filebasedsource
+from tensorflow_transform.tf_metadata import dataset_schema
+
+
+INPUT_SCHEMA = {
+    # Features (inputs)
+    'TotalC': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+    'TotalH': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+    'TotalO': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+    'TotalN': dataset_schema.ColumnSchema(
+        tf.int64, [], dataset_schema.FixedColumnRepresentation()),
+
+    # Labels (outputs/predictions)
+    'Energy': dataset_schema.ColumnSchema(
+        tf.float32, [], dataset_schema.FixedColumnRepresentation()),
+}
+
+LABELS = ['Energy']
+
+
+class ParseSDF(filebasedsource.FileBasedSource):
```

Comment:
Cham has been kind enough to explain this to me. I was right:

You will need to add an `__init__` method that calls its super-method with `splittable=False`. Per cham: "if not it will at least create a set of splits that read nothing, or lose data in the worst case".

Other than that, this LGTM.

Reply:
Update: I tried making it non-splittable and it failed. I sent an email for further investigation on this, but for the meantime I'm reverting it back to removing the constructor so it doesn't crash.

Reply:
Update 2: this is now non-splittable and is working :)